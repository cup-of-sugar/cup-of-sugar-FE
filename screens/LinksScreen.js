// import { Ionicons } from '@expo/vector-icons';
import * as WebBrowser from 'expo-web-browser';
import * as React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { RectButton, ScrollView } from 'react-native-gesture-handler';

export default function LinksScreen() {
  return (
    <ScrollView style={styles.container} contentContainerStyle={styles.contentContainer}>
      <OptionButton
        label="Read the Expo documentation"
        onPress={() => WebBrowser.openBrowserAsync('https://docs.expo.io')}
      />

      <OptionButton
        label="Read the React Navigation documentation"
        onPress={() => WebBrowser.openBrowserAsync('https://reactnavigation.org')}
      />

      <OptionButton
        // icon="ios-chatboxes"
        label="Ask a question on the forums"
        onPress={() => WebBrowser.openBrowserAsync('https://forums.expo.io')}
        isLastOption
      />
    </ScrollView>
  );
}

function OptionButton({ label, onPress, isLastOption }) {
  return (
    <RectButton style={[styles.option, isLastOption && styles.lastOption]} onPress={onPress}>
      <View style={{ flexDirection: 'row' }}>
        <View style={styles.optionTextContainer}>
          <Text style={styles.optionText}>{label}</Text>
        </View>
      </View>
    </RectButton>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fafafa',
  },
  contentContainer: {
    paddingTop: 15,
  },
  optionIconContainer: {
    marginRight: 12,
  },
  option: {
    backgroundColor: '#fdfdfd',
    paddingHorizontal: 15,
    paddingVertical: 15,
    borderWidth: StyleSheet.hairlineWidth,
    borderBottomWidth: 0,
    borderColor: '#ededed',
  },
  lastOption: {
    borderBottomWidth: StyleSheet.hairlineWidth,
  },
  optionText: {
    fontSize: 15,
    alignSelf: 'flex-start',
    marginTop: 1,
  },
});
