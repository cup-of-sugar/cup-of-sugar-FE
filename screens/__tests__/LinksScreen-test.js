import React from 'react';
import LinksScreen from '../LinksScreen';

import renderer from 'react-test-renderer';

test('should render component', () => {
  const tree = renderer.create(<LinksScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
