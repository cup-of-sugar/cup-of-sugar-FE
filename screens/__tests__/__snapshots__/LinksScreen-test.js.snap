// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
<RCTScrollView
  collapsable={false}
  contentContainerStyle={
    Object {
      "paddingTop": 15,
    }
  }
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  style={
    Object {
      "backgroundColor": "#fafafa",
      "flex": 1,
    }
  }
>
  <View>
    <RNGestureHandlerButton
      activeOpacity={0.105}
      collapsable={false}
      onActiveStateChange={[Function]}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      onPress={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "backgroundColor": "#fdfdfd",
              "borderBottomWidth": 0,
              "borderColor": "#ededed",
              "borderWidth": 0.5,
              "paddingHorizontal": 15,
              "paddingVertical": 15,
            },
            undefined,
          ],
        ]
      }
      underlayColor="black"
    >
      <View
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "alignSelf": "flex-start",
                "fontSize": 15,
                "marginTop": 1,
              }
            }
          >
            Read the Expo documentation
          </Text>
        </View>
      </View>
    </RNGestureHandlerButton>
    <RNGestureHandlerButton
      activeOpacity={0.105}
      collapsable={false}
      onActiveStateChange={[Function]}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      onPress={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "backgroundColor": "#fdfdfd",
              "borderBottomWidth": 0,
              "borderColor": "#ededed",
              "borderWidth": 0.5,
              "paddingHorizontal": 15,
              "paddingVertical": 15,
            },
            undefined,
          ],
        ]
      }
      underlayColor="black"
    >
      <View
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "alignSelf": "flex-start",
                "fontSize": 15,
                "marginTop": 1,
              }
            }
          >
            Read the React Navigation documentation
          </Text>
        </View>
      </View>
    </RNGestureHandlerButton>
    <RNGestureHandlerButton
      activeOpacity={0.105}
      collapsable={false}
      onActiveStateChange={[Function]}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      onPress={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "backgroundColor": "#fdfdfd",
              "borderBottomWidth": 0,
              "borderColor": "#ededed",
              "borderWidth": 0.5,
              "paddingHorizontal": 15,
              "paddingVertical": 15,
            },
            Object {
              "borderBottomWidth": 0.5,
            },
          ],
        ]
      }
      underlayColor="black"
    >
      <View
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "alignSelf": "flex-start",
                "fontSize": 15,
                "marginTop": 1,
              }
            }
          >
            Ask a question on the forums
          </Text>
        </View>
      </View>
    </RNGestureHandlerButton>
  </View>
</RCTScrollView>
`;
